### PR TITLE
Week1

### PR DESCRIPTION
**Pull request by HonzaDudek**

### CODE REVIEW

File: src/app/_helpers/InMemoryTokenService.ts [hunk 1]
```diff
@@ -6,20 +6,15 @@ import { Injectable } from '@angular/core';
 })
 export class InMemoryTokenService implements InMemoryDbService {
```
Comment (OndraSlejtr) [inline src/app/_helpers/InMemoryTokenService.ts]: We don't need this service anymore.

File: src/app/_models/token.interface.ts [hunk 2]
```diff
@@ -1,3 +0,0 @@
-export interface Token {
```
Comment (OndraSlejtr) [inline src/app/_models/token.interface.ts:1]: We definitely want to have interface models for our API responses.

File: src/app/app.component.spec.ts [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
 import { AppComponent } from './app.component';
+import {LoginComponent} from './login/login.component';
+import {RegisterComponent} from './register/register.component';
+import {BrowserModule} from '@angular/platform-browser';
+import {AppRoutingModule} from './app.routes';
+import {FormsModule} from '@angular/forms';
+import {HttpModule} from '@angular/http';
+import {HttpClientModule} from '@angular/common/http';
+import {HttpClientInMemoryWebApiModule} from 'angular-in-memory-web-api';
+import {InMemoryUsersService} from './_helpers/InMemoryUsersService';
```
Comment (OndraSlejtr) [inline src/app/app.component.spec.ts]: No longer needed.

File: src/app/app.component.spec.ts [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+        LoginComponent,
+        RegisterComponent
+      ],
+      imports: [
+        BrowserModule,
+        AppRoutingModule,
+        FormsModule,
+        HttpModule,
+        HttpClientModule,
+        HttpClientInMemoryWebApiModule.forRoot(InMemoryUsersService)
```
Comment (OndraSlejtr) [inline src/app/app.component.spec.ts]: No longer needed.

File: src/app/app.component.html [hunk 5]
```diff
@@ -10,6 +10,7 @@ <h2>Welcome to Tribes</h2>
   <p>Please <a routerLink="/login">login</a> or <a routerLink="/register">register</a>.</p>
   <router-outlet></router-outlet>
 </div>
+<a routerLink="/logout" (click)="logoutComponent.logout()">Logout</a>
```
Comment (OndraSlejtr) [inline src/app/app.component.html]: Simple reroute, no need for function call.

File: src/app/app.component.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import {Component, OnInit, ViewChild} from '@angular/core';
+import {LogoutComponent} from './logout/logout.component';
```
Comment (OndraSlejtr) [inline src/app/app.component.ts]: Unnecessary, as pointed above - simple send user to route, no need to hold logout component reference.

File: src/app/app.component.ts [hunk 7]
```diff
@@ -9,5 +10,5 @@ export class AppComponent {
   title = 'Vulpes|Tribes';
   @ViewChild('formElement') signupForm;
 
-
+  constructor(private logoutComponent: LogoutComponent) {}
```
Comment (OndraSlejtr) [inline src/app/app.component.ts]: Unnecessary, as pointed above - simple send user to route, no need to hold logout component reference.

File: src/app/login/login.component.ts [hunk 8]
```diff
@@ -23,8 +24,10 @@ export class LoginComponent implements OnInit {
   onSubmit() {
     this.user.username = this.loginForm.value.username;
     this.user.password = this.loginForm.value.password;
+    console.log(this.loginService.isLoggedIn());
```
Comment (OndraSlejtr) [inline src/app/login/login.component.ts]: We can keep this stuff in development branch, but for that, we need to create checked for whether we are deploying in prod or dev version (so it doesn't print debug info into user's console).

File: src/app/login/login.service.ts [hunk 9]
```diff
@@ -1,21 +1,34 @@
...
+  }
 
   login(user) {
-    this.http.post('login', user);
-    this.token.saveToken();
+    this.http.post<any>(environment.login, user)
+      .subscribe(
+        response => {
+          console.log(response.tribes_token);
+          this.saveToken(response.tribes_token);
```
Comment (OndraSlejtr) [inline src/app/login/login.service.ts:22]: User should be rerouted after login is finished. Meanwhile, you can display the "progress" in Login Component (turning wheel, moving circle etc)

File: src/app/logout/logout.component.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export class LogoutComponent implements OnInit {
+
+
+  constructor(private logoutService: LogoutService) { }
+
+  ngOnInit() {
+  }
+
+  logout() {
+    this.logoutService.logout();
```
Comment (OndraSlejtr) [inline src/app/logout/logout.component.ts:16]: It should reroute user to root.

File: src/app/register/register.component.ts [hunk 11]
```diff
@@ -7,29 +8,23 @@ import {Component, OnInit, ViewChild} from '@angular/core';
...
 
-  constructor() { }
+  constructor(private registerService: RegisterService) { }
 
   ngOnInit() {
   }
 
   onSumbit() {
-    this.user.id = Math.floor(Math.random() * (20000 - 10000));
     this.user.username = this.registerForm.value.username;
```
Comment (OndraSlejtr) [inline src/app/register/register.component.ts]: Binding should be used here, instead of manual setting (since we are using Template driven forms).

Comment (OndraSlejtr) [reply src/app/register/register.component.ts]: This also means change to template.

**Review by OndraSlejtr — DISMISSED** [review]

File: src/app/login/login.service.ts (hunk 9 above)
Comment (HonzaDudek) [reply src/app/login/login.service.ts:22]: Waiting on /game component for rerouting

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**